dash_labs/plugins/pages.py

```diff
@@ -398,3 +430,24 @@ def _parse_query_string(search):
 
         parsed_qs[k] = first
     return parsed_qs
+
+
+def _parse_path_variables(pathname, path_template):
+    path_segments = pathname.split("/")
+    template_segments = path_template.split("/")
+
+    if len(path_segments) != len(template_segments):
+        return None
+
+    path_vars = {}
+    for i in range(len(path_segments)):
```

Comment:
this would be a little cleaner as:
`for path_segment, template_segment in zip(path_segments, template_segments):`
since you already know the lengths are the same.

That said I suspect at some point we'll want to extend this to support [Flask converter types](https://flask.palletsprojects.com/en/2.0.x/quickstart/#variable-rules) including `path` that also allows slashes in the match, but that will need some special logic anyway.

Reply:
I thought about doing the zip, but then there would be additional logic to pull out the variables

Reply:
Where? All I see is `path_segments[i]` which would just turn into `path_segment`, and similarly with `template_segments[i]`->`template_segment`. You never use `i` for any purpose other than pulling out the ith element of those two lists, so you should be able to get rid of it entirely.

Reply:
I have a new version [here](https://github.com/plotly/dash-labs/pull/61/commits/9c5e92bc6cd5c9c4437b1a64face96ea085bc5a2) is it better?

I also removed the conversion from "None" to `None`.   

Also, When no `path` is supplied to  `dash.page_registry` I changed the path variables in the inferred path  "none"  (instead of "None").  I hope this makes it more clear that it's a string and not actually `None`